Counters

TomasBakos opened the pull request:
Added move and time counters with printing on console.

TomasBakos committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into save/load

TomasBakos committed:
add task card about counters

TomasBakos committed:
add move and time counter

filipKov reviewed (changes requested):
`MILIARD` => `BILLION` pls

TomasBakos committed:
change milion to billion as requested

mitas1 commented on PuzzleSlider/src/Engine/GameData.java:
```diff
@@ -40,6 +44,10 @@ public String toString() {
 			out.append( Arrays.toString(mTiles[i]));
 			out.append(System.lineSeparator());
 		}
+		out.append("Pocet tahov: " + mMoveCount);
```
Is there any reason for being inconsistent here? I mean, we are using english?. And, I hate java but is this really suitable in java to add `integer` to `string` whitout any formating function or sth like that?

TomasBakos replied:
Didnt put much thought to which language im using, since it was added just for testing purposes. Additionally our application is going to have non-console design, so these contents of toString() fuction will be irrelevant at the end of the sprint.
To the second point. I dont really know. I've been using that long time and didnt see any problem there.
I can rework that if you insist, but i think it's irrelevant.

mitas1 replied:
So, yes.Please, rewrite to english. Thanks. And to the second point, I dont know if it is irrellevant in java, so it is your choice ;)

mitas1 commented on PuzzleSlider/src/Engine/GameData.java:
```diff
@@ -40,6 +44,10 @@ public String toString() {
 		}
+		out.append("Pocet tahov: " + mMoveCount);
+		out.append(System.lineSeparator());
+		out.append("Cas: " + (getTimeInSeconds() - mStartTime) + " sekund");
```
The same. Inconsitency.

mitas1 commented on PuzzleSlider/src/Engine/GameData.java:
```diff
@@ -124,6 +137,23 @@ public GridPoint getEmptyTile() {
+	
+	public int incrementMoveCount(){
+		mMoveCount++;
+		return mMoveCount;
```
Why are you returning `mMoveCount` here? I do not see it is being used and you have getter for this protected variable.

TomasBakos replied:
The idea was maybe in future it would be handy to immediately return incremented value and not use incerement and getter functions.

mitas1 replied:
As our master Gyarfas always saying "What will be in the future ..." So rewrite ;)

mitas1 commented on PuzzleSlider/src/Engine/Engine.java:
```diff
@@ -20,6 +20,7 @@ public Engine(int size) throws InvalidArgumentException {
 		mGameData = new GameData(size);
 		mGameData.initialize();
 		shuffleNewGame(1000);
+		mGameData.initVariables();
```
Why are you initializing variables outside of class? Any time you are creating new instance of class you have to remember that. I hope there is better way to that in `initialize()` for example ... Or am I wrong?

TomasBakos replied:
Maybe just misused function name. Should be more like "initCounters()". The problem here is, we can't initialize them in "initialize()" function because the "suffleNewGame" function is then interfering with counters i.e. they are counting moves and time during shuffling new game.

mitas1 replied:
`initCounters()` sounds better

TomasBakos committed:
reworked in accordance to the latest comments

Language consistency
Changed return value of function incrementMoveCount() to void
Renamed  function initVariables() to initCounters()
